src/main/java/uk/gov/hmcts/reform/blobrouter/data/reconciliation/reports/ReconciliationReportRepository.java

```diff
@@ -87,4 +87,15 @@ public UUID save(NewReconciliationReport report) throws SQLException {
             return Optional.empty();
         }
     }
+
+    public void updateDetailedContent(UUID id, String newDetailedContent) throws SQLException {
+        jdbcTemplate.update(
+            "UPDATE envelope_reconciliation_reports "
+                + "SET detailed_content = :newDetailedContent "
+                + "WHERE id = :id",
+            new MapSqlParameterSource()
+                .addValue("id", id)
+                .addValue("newDetailedContent", toJson(newDetailedContent))
+        );
+    }
```

Comment:
it would be nice to add a test checking that only valid json is accepted

Reply:
added